playground/backend/internal/code_processing/code_processing.go

```diff
@@ -57,10 +56,11 @@ func Process(ctx context.Context, cacheService cache.Cache, lc *fs_tool.LifeCycl
 
 	go cancelCheck(ctxWithTimeout, pipelineId, cancelChannel, cacheService)
 
-	// build executor for validate and compile steps
-	executor := compileBuilder.Build()
-
-	// validate
+	// Validate
+	executor, err := builders.SetupValidator(lc.GetAbsoluteSourceFilePath(), sdkEnv.ApacheBeamSdk)
+	if SetupError(err, pipelineId, cacheService, ctxWithTimeout) {
```

Comment:
May be it will be better to use something like this:
```
executor, err := builders.SetupValidator(lc.GetAbsoluteSourceFilePath(), sdkEnv.ApacheBeamSdk)
if err != nil {
    processSetuperrError(err, pipelined, cacheService, ctxWithTimeout)
    return
}
``` 

And `processSetuperrError()` will only log the error:
```
logger.Errorf("%s: error during setup builder: %s\n", pipelineId, err.Error())
cacheService.SetValue(ctxWithTimeout, pipelineId, cache.Status, pb.Status_STATUS_ERROR)
```

Reply:
Agree, I think, this way it will be more explicit